README.md

```diff
@@ -139,6 +139,36 @@ Notes:
 
 As a result, you should see Dex deployed in your workload cluster.
 
+### Ingress, TLS and custom certification authorities
+
+Dex app exposes a web interface, which is accessible over https. Therefore, it creates an ingress, which needs to be configured with a TLS certificate signed by a certification authority, which needs to be trusted by the browsers. 
+The app consists of several components, which also need to be able to communicate with each other internally over https. So the certification authority signing the certificates needs to be trusted by the individual app components as well.
+
+In case a custom certification authority is used, it needs to be exposed to the individual app components and set as trusted, otherwise the components will not be able to communicate with each other and the app may not work as expected.
+Based on the cluster setup, this can be achieved by providing an additional set of values to the app configuration:
+
+1. Add a base64-encoded certificate of the certification authority to the User Values configmap or secret. This option is useful when using custom, self-signed certificates in a cluster:
+
+```yaml
+ingress:
+  tls:
+    letsencrypt: false
+    caPemB64: "base64-encoded CA certificate"
+```
```

Comment:
Is my understanding correct that the PEM data is supposed to be base64-encoded in addition?

PEM data is normally easy to add as a multi line value in a YAML file. We do this in several places. I see no need for additional base64 encoding. In fact I think additional encoding would make this harder to handle and would introduce another way to make a mistake.

If I'm misunderstanding this, then maybe it's unclear.

Another thing: CA certificates are not "secrets". I would not even suggest to put them into a Secret resource.

Reply:
Regarding the CA being stored in a secret - it is beyond the scope of this PR to decide which type of resource should be used to store a certification authority. The decision has already been made by the cluster implementation. The CA secret already exists in the cluster and various components and services already use it. And Dex needs to be able to use it, too.